### ISSUE TITLE
 Incorrectly triggering dragenter and dragleave events in v-for loop - continuation to #4479 

### ISSUE THREAD

**Issue by Eggon-666** [vuejs/vue#10061]
### Version
2.6.10

### Reproduction link
[https://jsfiddle.net/Eggon666/d6htzpr0/12/](https://jsfiddle.net/Eggon666/d6htzpr0/12/)

### Steps to reproduce
Drag the buttons (on Windows, Chrome). Watch console.log reports on number of button on which dragenter/dragleave event was triggered - for buttons 3-5 the number might be incorrect. The bug sometimes needs a few drags before it triggers. 

### What is expected?
First dragenter/dragleave event triggering inside the dragged element should trigger only on the element that is being dragged (unless the pointer moves over another element).

### What is actually happening?
**Dragenter** and **dragleave** events are triggered on an incorrect (i.e. **preceding sibling**) element from **v-for** loop **together with or instead of** the events of the actually dragged element (DragTarget). The problem seems to be related to v-for specifically, I haven’t encountered it on non-v-for’ed elements. It appears on Chrome only.

Bug behavior is erratic. There are 5 elements rendered with v-for:

1) The problem does not always occur on the first drag

2) Buttons 1 and 2 always work fine

3) Button 3 sometimes works fine and sometimes does not trigger any dragenter/dragleave event at all

4) Buttons 4 and 5:

  a.	sometimes work fine (rarely),

  b.	sometimes they trigger events only on the preceding element (eg dragging No 4 triggers dragenter/leave on No 3)

  c.	sometimes they trigger events on both the dragged element and the preceding element. In such case the order is: 1) PrecedingSibling.DragEnter 2) DragTarget.DragEnter 3) PrecedingSibling.DragLeave 4) DragTarget.DragLeave

  d.	However, if you start the drag and don’t move the pointer outside of the DragTarget the bug is **very unlikely** to occur! If it occurs it never triggers PrecedingSibling only, it triggers both elements: 1) PrecedingSibling.DragEnter 2) DragTarget.DragEnter 3) PrecedingSibling.DragLeave (the latter is therefore not connected with DragTarget.DragLeave, but happens automatically regardless) 

  e.	On the other hand, if you move the pointer out of the element immediately after drag start, the bug is very likely to occur.

  f.	When the dragenter is triggered on the DragTarget again, i.e. when you move the pointer out of the element and come back into it – the sibling event is not triggered anymore, just on the very beginning of the drag.

Bug is present on all 3 versions of Vue in js fiddle and on Vue 2.6.10 in my project.

---
This is the same problem as was submitted in unresolved issue **#4479**. I can reproduce the behavior from the jsfiddle mentioned in #4479 on my PC. I hope this submission can shed some more light on the problem, because I noticed some more nuances.

Bug behavior on my local project (Vue 2.6.10) is slightly different than the one from my fiddle described above. The setup is slightly different as well (v-for is used on components which have some nested divs in them). Bug behavior:

1) The problem does not always occur on the first drag

2) Once the problem occurs for the first time it occurs on every consecutive drag – except for element no 2 which can behave correctly again.

3) Dragging element triggers dragenter/dragleave triggers on both the dragged element and on the preceding v-for sibling (sibling first) or just on the preceding sibling. When the dragenter is triggered on the DragTarget again, i.e. when you move the pointer on the child div or when you leave the element and come back into it – the sibling event is not triggered anymore, just on the beginning of the drag.

The difference between js fiddle and local behavior is mainly (but not only) frequency of particular behaviors. On the local setup:

a. second v-for element is also affected, although can react correctly sometimes (in fiddle second is not affected),

b. third element very rarely ignores the drag (in fiddle it is much more frequent)

c. starting from third element rendered by v-for, once the bug occurs, it will occur every time (although will still randomly oscillate between triggering on preceding sibling only or two elements together),

d. the fact of not moving the cursor out of the DragTarget does not prevent the bug from happening (in contrary to jsfiddle description point 4.d ). This might be caused by children divs?

Problem occurs on my two computers (Win7 HP and Win7 Professional), only on Chrome (74.0.3729.157). Mozilla and IE work fine. On one of the computers I removed Chrome with all extensions and history and reinstalled it. The problem persisted. Kaspersky’s Chrome extension is unfortunately automatically loaded on Chrome install, though not activated. Turning off Kaspersky on Windows did not prevent the extension from loading, and I don’t know how to prevent it other than uninstalling Kaspersky which I’d rather not do. Removing the Kaspersky’s extension from Chrome does not change the situation though.

PS. This is my second submission of this issue. Previous was closed by the bot, because I used "reference in new issue" option from my comment to #4479. I thought that linking them this way would be more useful than going through the template. Please consider adding explicit clarification on how to submit issues "reopening" previously closed issues  and/or a issue reference field to the template. I read the documentation, but didnt find it. Sorry for inconvenience.

**Comment by ZY2071** [comment]
I swear it's the most detailed issue I have ever seen.

**Comment by ZY2071** [comment]
> I swear it's the most detailed issue I have ever seen.

but I can't reproduce the bug too,Maybe It's not a bug about vue.

**Comment by Eggon-666** [comment]
Haha, hope that's a good thing though. :) 
For any future persons seeking answer I was able to prove this is not Vue indeed. I reproduced the bug without any Vue involvement in this fiddle:
https://jsfiddle.net/Eggon666/jd4ztfyh/6/

I thought it was Vue, because it started appearing when I rewrote my project to Vue from vanilla JS and in the vanilla code I didnt experience it (I did check it obviously). Also when I googled for an answer Vue issue #4479 came up and no "general js" bugs, so it looked too obvious. Sorry for the trouble. At least it's clear now Vue's innocent.
Cheers!
